build.assets/Dockerfile

```diff
@@ -159,6 +159,15 @@ RUN apt-get -y update && \
         xauth \
         libpcsclite-dev \
         && \
+    sudo install -m 0755 -d /etc/apt/keyrings && \
+    curl -fsSL https://download.docker.com/linux/ubuntu/gpg | sudo gpg --dearmor -o /etc/apt/keyrings/docker.gpg && \
```

Comment:
I feel like this really defeats the purpose of signing packages - the same host that holds the key holds the packages we are verifying. If someone hacks download.docker.com with a dodgy package, they can just overwrite the gpg public key too.

We should download the key, verify it and put it in our repo to copy into the docker image instead of downloading it each time.

Reply:
Addressed in 7eec8f8860f7ae651fcac8b4facbf7e6f2f4e919